tests/test_pydss_examples.py

```diff
@@ -0,0 +1,150 @@
+
+import datetime
+import os
+import re
+import shutil
+import tempfile
+from distutils.dir_util import copy_tree
+
+import pandas as pd
+import pytest
+import subprocess
+from PyDSS.common import PROJECT_TAR, PROJECT_ZIP
+from PyDSS.exceptions import InvalidParameter
+from PyDSS.pydss_fs_interface import PROJECT_DIRECTORIES, SCENARIOS, STORE_FILENAME
+from PyDSS.pydss_project import PyDssProject, PyDssScenario, DATA_FORMAT_VERSION
+from PyDSS.pydss_results import PyDssResults, PyDssScenarioResults
+from tests.common import RUN_PROJECT_PATH, SCENARIO_NAME, cleanup_project
+
+
+PATH = os.path.join(tempfile.gettempdir(), "pydss-projects")
+EXAMPLES_path = "examples"
+
+@pytest.fixture
+def pydss_project():
+    if os.path.exists(PATH):
+        shutil.rmtree(PATH)
+    yield
+    if os.path.exists(PATH):
+        shutil.rmtree(PATH)
+
+
+base_projects_path = None
+def copy_examples_to_temp_folder():
+    #assert os.path.exists(EXAMPLES_path)
+    proc = None
+    base_projects_path = os.path.join(PATH, "pydss_projects")
+    print(f"Temporary path: {base_projects_path}")
+    os.makedirs(base_projects_path, exist_ok=True)
+    assert os.path.exists(base_projects_path)
+    copy_tree(EXAMPLES_path, base_projects_path)
+    return base_projects_path
+
+def test_external_interfaces_example():
+    example_name = "external_interfaces/pydss_project"
+    scenarios = [
+        {
+            'TOML': 'helics.toml',
+            'file': r"external_interfaces\Helics_example\Federate_runner.bat",
```

Comment:
`\` won't work on Linux and Mac.  You can always use `/`.

Anyway, these .bat files won't run on Linux or Mac.  Need to detect if Windows or not.  You can use `sys.platform`.  Can this be handled in a way that is platform agnostic?  You could use .bat for Windows and .sh for the others.  Or you could make these Python scripts.

Reply:
I will update this bit tomorrow. Will try and run the python file directly.

Reply:
Done. tests now use python files rather than